src/DynamoCore/Models/NodeModel.cs

```diff
@@ -192,6 +192,42 @@ public bool IsUpstreamVisible
         }
 
         /// <summary>
+        /// Input nodes are used in Customizer and Presets. Input nodes can be numbers, number sliders,
+        /// strings, bool, code blocks and custom nodes, which don't specify path.
+        /// </summary>
+        public bool IsInputNode
+        {
+            get
+            {
+                return !inPorts.Any() && !(this is DSFunction);
+            }
+        }
+
+        // By default node is visible in Customizer.
+        // Even if it's not input node it will be visible.        
+        private bool isSelectedInput = true;
+        /// <summary>
+        /// All input nodes can be used in customizer or be part of preset.
+        /// But, if user would like, he can hide input node from customizer.
+        /// By default input node will be selected, so that it can be used in preset and be shown in customizer.
+        /// </summary>
+        public bool IsSelectedInput
```

Comment:
Let's keep the references to customizer or publishing out of DynamoCore.  They are, strictly speaking, extensions to Dynamo.  

"Specifies whether an input node can be included in a preset.  By default, this field is set to true."